server/graphql/invoice/src/mutations/outbound_return/delete.rs

```diff
@@ -30,26 +36,35 @@ pub enum DeleteResponse {
     Response(DeletedIdsResponse),
 }
 
-pub fn delete(_ctx: &Context<'_>, _store_id: &str, _input: DeleteInput) -> Result<DeleteResponse> {
-    // let user = validate_auth(
-    //     ctx,
-    //     &ResourceAccessRequest {
-    //         resource: Resource::MutateOutboundShipment,
-    //         store_id: Some(store_id.to_string()),
-    //     },
-    // )?;
+pub fn delete(ctx: &Context<'_>, store_id: &str, input: DeleteInput) -> Result<DeleteResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            // resource: Resource::MutateOutboundReturn,
+            resource: Resource::MutateOutboundShipment,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
 
-    // let service_provider = ctx.service_provider();
-    // let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
+    let service_provider = ctx.service_provider();
+    let service_context = service_provider.context(store_id.to_string(), user.user_id)?;
 
-    // map_response(
-    //     service_provider
-    //         .invoice_service
-    //         .delete_inbound_shipment(&service_context, input.to_domain()),
-    // )
-    Ok(DeleteResponse::Response(DeletedIdsResponse(vec![
-        "deleted_id".to_string(),
-    ])))
+    map_response(
+        service_provider
+            .invoice_service
+            .delete_outbound_returns(&service_context, input.ids),
```

Comment:
This is a bit different than the pattern elsewhere - I'm accepting a vec of ids to delete, and then looping in the service... this felt more direct than the batchMutation deletes done elsewhere. Keen for feedback on this?

I suppose the other option would be to make this a singular `deleteOutboundReturn` mutation, and call it multiple times from the frontend?

Reply:
@mark-prins and I were discussing this too.

Batch mutation deletes feels a bit heavy going. We're planning to do the single delete mutation called multiple times (similar to what we did in notify). I think the times when we want to delete a lot of items should be very rare, so in most cases there's not going to be a big performance improvement, or even if there is, is it important?

Would be good to discuss as a team and agree on a pattern going forward though.

Reply:
this would be my preference - though we've just implemented the 'loop in the front end' version in assets.
one question - what happens with errors? if one of the deletes fail, is this a transactional operation now, so the entire Vec of ids is returned as an error? I think the batch mutation returns a collection of responses, so you can have some fail and some succeed